### ISSUE TITLE
feat: expose promise for verify, run, etc.

### ISSUE THREAD

**Issue by AndersDJohnson** [mockyeah/mockyeah#232]
We could expose a promise for the async verify, the `run` registration, etc., as an alternative to registering a `done` handler, etc. This could be useful in some test frameworks that support returning promises and/or `async`/`await`, such as Jest. Examples:

```js
test('should do something', () =>
  mockyeah
      .get('/foo', { text: 'bar' })
      .expect()
      .params({
        id: '9999'
      })
      .once()
      .run(() => fetch('/foo?id=9999'))
      .verify()
);
```

**Comment by AndersDJohnson** [comment]
Fixed by #271.

**Comment by AndersDJohnson** [comment]
#271 did not expose a promise
